### PR TITLE
Remove allocs from NuGetv3LocalRepository.FindPackage

### PR DESCRIPTION
**Pull request by benaadams**
Removes 126 MB of IEnumerators in linked issue

Resolves https://github.com/NuGet/Home/issues/5679

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol/PackagesFolder/NuGetv3LocalRepository.cs [hunk 1]
```diff
@@ -157,7 +171,7 @@ public void ClearCacheForIds(IEnumerable<string> packageIds)
                 // Clearers must wait for all requests to complete
                 lock (GetLockObj(packageId))
                 {
-                    IEnumerable<LocalPackageInfo> packages;
+                    List<LocalPackageInfo> packages;
                     _cache.TryRemove(packageId, out packages);
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol/PackagesFolder/NuGetv3LocalRepository.cs]: nit: `out var packages`

Comment (alpaix) [reply src/NuGet.Core/NuGet.Protocol/PackagesFolder/NuGetv3LocalRepository.cs]: or, to be even more concise, `out var _`

Comment (benaadams) [reply src/NuGet.Core/NuGet.Protocol/PackagesFolder/NuGetv3LocalRepository.cs]: Seems you don't even need the var for the don't care `_`

**Review by alpaix — APPROVED** [review]
Thanks for the PR!
Waiting for the test pass.

**Review by nkolev92 — APPROVED** [review]

**Review by emgarten — APPROVED** [review]
LGTM, thanks for the fix! 🚀

**Pull request merged**